hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/transaction/ConcurrentOperation.java

```diff
@@ -84,6 +86,38 @@ public ConcurrentOperation(HoodieInstant instant, HoodieCommitMetadata commitMet
     init(instant);
   }
 
+  /**
+   * Construct a ConcurrentOperation object for a compaction plan that is not yet persisted to the timeline.
+   * This is used to detect potential conflicts with other operations.
+   * @param compactionPlan the potential compaction plan
+   * @param requestedInstantTime the requested instant time for the compaction
+   */
+  public ConcurrentOperation(HoodieCompactionPlan compactionPlan, String requestedInstantTime) {
+    this.commitMetadataOption = Option.empty();
+    this.metadataWrapper = null;
```

Comment:
Let's move the `metadataWrapper` to be a local variable used only for `#init` purposes.

Reply:
Makes sense, updated the PR